### PR TITLE
Prefix trie

### PR DESCRIPTION
**Pull request by QuinnWass**
Please review construction and insertion logic. Will do search tomorrow.

### CODE REVIEW

**Comment by QuinnWass** [comment]
Unit tests pass

**Review by galaxyonfire — DISMISSED** [review]
Seems that unit tests fail to run when compiled on cs1.
image
Think it's an issue on main. I created a fix from dealing with this error on the reorg branch which is to change airport_server:84 to `if (line.size() > 1)` cause I did some testing and for some reason the std:getline on cs1 reads empty lines as having a length of 1 (think it may be because there is a newline char in it).

File: mergedFiles/prefix_trie_unit_tests.c [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        places_origin origin = p.findLocation("AXDYZ");
+    }
+    catch(const std::out_of_range& e)
+    {
+        return;
+    }
+    assert(false);
+}
+
+void run_prefixTrie_tests()
```
Comment (galaxyonfire) [inline mergedFiles/prefix_trie_unit_tests.c:218]: Do you want to add a test for handling the ambiguous case?
Ex: test_a = "ABC", test_b = "AXY" -> input is "A" -> exception is thrown

**Review by Arjunica — APPROVED** [review]
looks good. Approving

**Pull request merged**